### DISCUSSION TITLE
"Cmd" for toogling presentation using Tabs

### DISCUSSION THREAD

**Discussion by moussaamrani** [sumatrapdfreader/sumatrapdf#3408] [Ideas]
Thanks for this amazing tool that I use daily for my work, and for my personal life (mangas, hey! :-) )

I wanted to ask whether it would be possible to have a specific `CmdToogleTabView` (or something of the like) that would correspond to the "_Use Tabs_" in the _Options_.

This would allow to associate it with a shortcut to switch views, depending on specific usage of Sumatra:
- when several files need to be read, having the tabs available for each new opened file is very practical;
- on the other hand, disabling the use of tabs is useful in other situation where PDF files are opened only occasionally and the previously opened files staying there is not needed.

Thanks a lot!

M.A

**Comment by GitHubRulesOK** [comment]
unclear  implementation ?
For toggle between tabs and full screen there is <kbd>F</kbd> and several others
but the whole reason for the tabs feature is to have several current views stacked

If you want only sub sets for different times that's a user custom ability provided by externally applied -appdata  switch so say one set for comics one set for bills. see https://forum.sumatrapdfreader.org/t/sessions-in-sumatra/1272 or in your case an `-appdata notabs` folder.

![image](https://user-images.githubusercontent.com/12754913/229120992-5483f299-13df-4c07-84a6-57b55c0fbcf4.png)



To store a historic set for later use, that need some work and is doable by run cmd script to say backup current collection. Clearly an already suggested custom feature see https://github.com/sumatrapdfreader/sumatrapdf/discussions/3045

**Reply by moussaamrani** [reply]
Sorry, maybe my use case was not clear enough. So you may as well forget about it.

However, the beginning of my request is a bit clearer: is there a "command" (I suggested `CmdToogleTabView`, but any other convenient name will do) that would mimic a user clicking or unclicking the "Use Tabs" entry in _`Settings > Options > Use Tabs`_ ?
This would be similar to, for example, 
- the `c` shortcut that mimics the user clicking/unclicking _`View > Show Pages Continually`_; or
- closer to where the options that interests me ("Use Tabs"), there exists the CmdCheckUpdate to mimic clicking/unclicking _`Settings > Options > Automatically Check For Updates`_.

I am just surprise the command for checking updates exist, while the one for using tabs does not... Unless it is not listed at https://www.sumatrapdfreader.org/docs/Commands. I also checked the COMMAND(V) function definition at https://github.com/sumatrapdfreader/sumatrapdf/blob/master/src/Commands.h#L9, where there is also no trace of it).

Hope it's more clear!

M.A

**Comment by GitHubRulesOK** [comment]
Toggle use tabs requires a restart, however there is a quirk (personally will occasionally make use off) when done "real time" and traditionally is set/held with other more Primary/Basic Start-up Options,
![image](https://user-images.githubusercontent.com/12754913/229117853-3c255a01-4eb2-432b-bf5b-04fa276d1491.png)


There would be a lot of re-setting to try and do similar "on the fly" as a toggle, possibly not impossible, but historically will be done at time of starting.

**Comment by kjk** [comment]
It's too much of a UI change to make it easily switchable. You can set `NoHomeTab` advanced option to have tab mode behave almost like no tabs and now you can easily drag out tabs to new windows.
